GeoLib/Point.h

```diff
@@ -27,27 +27,31 @@ namespace GeoLib
  * \ingroup GeoLib
  */
 
-class Point : public MathLib::Point3d, public GeoLib::GeoObject
+class Point : public MathLib::Point3dWithID, public GeoLib::GeoObject
 {
 public:
-	Point(double x1, double x2, double x3) :
-		MathLib::Point3d(), GeoLib::GeoObject()
-	{
-		this->_x[0] = x1;
-		this->_x[1] = x2;
-		this->_x[2] = x3;
-	}
+	Point(double x1, double x2, double x3,
+		std::size_t id = std::numeric_limits<std::size_t>::max()) :
+		MathLib::Point3dWithID(std::array<double,3>({{x1, x2, x3}}), id),
+		GeoLib::GeoObject()
+	{}
 
 	Point() :
-		MathLib::Point3d(), GeoLib::GeoObject()
+		MathLib::Point3dWithID(), GeoLib::GeoObject()
+	{}
+
+	Point(double const* x) :
+		MathLib::Point3dWithID(std::array<double,3>({{x[0], x[1], x[2]}}), 0),
+		GeoLib::GeoObject()
 	{}
 
-	Point (double const* x) :
-		MathLib::Point3d(x), GeoLib::GeoObject()
+	Point(Point const& p, std::size_t id = std::numeric_limits<std::size_t>::max()) :
```

Comment:
You are right. I will remove the default value for the second argument. Then the compiler generates its own copy constructor.